src/CharacterState.js

```diff
@@ -98,41 +98,48 @@ export default class CharacterState {
         let newYPos = 0;
         let newXPos = 0;
         let turnSegment = [];
-
-        switch(this.sceneDimensions.getBoundsStateX(this.xPos + xOffset)) {
-            case 'outOfBoundsBelow':
-                newXPos = 1;
-                if (!this.sceneDimensions.isSceneEdgeY(this.yPos)) {
-                    turnSegment = this.drawEdgeDiagonalPath([], this.xPos - newXPos, this.direction);
-                }
-                break;
-            case 'outOfBoundsAbove':
-                newXPos = this.sceneDimensions.getWidth();
-                if (!this.sceneDimensions.isSceneEdgeY(this.yPos)) {
-                    turnSegment = this.drawEdgeDiagonalPath([], newXPos - this.xPos, this.direction);
-                }
-                break;
-            default:
-                newXPos = this.xPos + xOffset;
-                break;
-        }
-
-        switch(this.sceneDimensions.getBoundsStateY(this.yPos + yOffset)) {
-            case 'outOfBoundsBelow':
-                newYPos = 1;
-                if (!this.sceneDimensions.isSceneEdgeX(this.xPos)) {
-                    turnSegment = this.drawEdgeDiagonalPath([], this.yPos - newYPos, this.direction);
-                }
-                break;
-            case 'outOfBoundsAbove':
-                newYPos = this.sceneDimensions.getHeight();
-                if (!this.sceneDimensions.isSceneEdgeX(this.xPos)) {
-                    turnSegment = this.drawEdgeDiagonalPath([], newYPos - this.yPos, this.direction);
-                }
-                break;
-            default:
-                newYPos = this.yPos + yOffset;
-                break;
+        const boundsStateX = this.sceneDimensions.getBoundsStateX(this.xPos + xOffset);
+        const boundsStateY = this.sceneDimensions.getBoundsStateY(this.yPos + yOffset);
+
+        if (boundsStateX === 'outOfBoundsBelow' && boundsStateY === 'outOfBoundsBelow') {
```

Comment:
I think this could be expressed more cleanly, perhaps by splitting up the logic by X and Y.  For example, in all cases where `boundsStateX` is `outOfBoundsBelow`, `newXPos` gets set to `1`, and yet you set it three times.

Reply:
We could group all the cases where boundsStateX is the same, but that will introduce condition checks for different boundStateY inside of the condition, which may make reading the code more difficult. I think writing all possible combinations with some redundancy has better readability.